File: app/src/main/java/com/iita/akilimo/models/ProducePrice.java
package com.iita.akilimo.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.orm.SugarRecord;

import java.util.Date;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class ProducePrice extends SugarRecord<ProducePrice> {

    @JsonIgnore
    long id;

    @JsonProperty("priceIndex")
    long priceIndex;

    @JsonProperty("priceId")
    long priceId;

    @JsonProperty("country")
    String country;

    @JsonProperty("minLocalPrice")
    double minLocalPrice;

    @JsonProperty("maxLocalPrice")
    double maxLocalPrice;

    @JsonProperty("minUsd")
    double minUsd;

    @JsonProperty("maxUsd")
    double maxUsd;

    @JsonProperty("active")
    boolean active;

    @JsonProperty("averagePrice")
    double averagePrice;

    @JsonProperty("createdAt")
    Date createdAt;

    @JsonProperty("updatedAt")
    Date updatedAt;
}

File: app/src/main/java/com/akilimo/mobile/models/ProducePrice.kt
package com.akilimo.mobile.models


import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
open class ProducePrice {
    @JsonProperty("priceIndex")
    var priceIndex: Long = 0

    @JsonProperty("priceId")
    var priceId: Long = 0

    @JsonProperty("country")
    var country: String? = null


    @JsonProperty("countryPrice")
    var countryPrice: String? = null

    @JsonProperty("minLocalPrice")
    var minLocalPrice = 0.0

    @JsonProperty("maxLocalPrice")
    var maxLocalPrice = 0.0

    @JsonProperty("minUsd")
    var minUsd = 0.0

    @JsonProperty("maxUsd")
    var maxUsd = 0.0

    @JsonProperty("active")
    var active = false

    @JsonProperty("averagePrice")
    var averagePrice = 0.0

    @JsonProperty("createdAt")
    var createdAt: Date? = null

    @JsonProperty("updatedAt")
    var updatedAt: Date? = null
}
